### PR TITLE
update communications page of cookbook to python3

### LINKED ISSUE #3472: Update communications page of cookbook
## Description

Currently the [communications page](https://docs.qgis.org/testing/en/docs/pyqgis_developer_cookbook/communicating.html) of the cookbook warns it is out of date. PR https://github.com/qgis/QGIS-Documentation/pull/3471 updates and tests the code examples.

## Checklist

- [x] I'm aware this repository is about QGIS Documentation and the issue I'm reporting is related to its usage.
If it's related to QGIS application or website, please refer to http://qgis.org/en/site/getinvolved/development/bugreporting.html
for the right issue tracker or to the [support channels](http://qgis.org/en/site/forusers/support.html)
- [x] I have added a link to the page concerned by the issue

### PR DESCRIPTION
**Pull request by ianturton**
Fixes #3472 - check and where necessary fix the code and text to reflect changes to Pythion 3 and QGIS 3.

### CODE REVIEW

File: source/docs/pyqgis_developer_cookbook/communicating.rst [hunk 1]
```diff
@@ -42,7 +40,7 @@ You can set a duration to show it for a limited time
 
 .. code-block:: python
 
-    iface.messageBar().pushMessage("Error", "Ooops, the plugin is not working as it should", level=Qgis.Critical, duration=3)
+    iface.messageBar().pushMessage("Notice", "Ooops, the plugin is not working as it should", level=Qgis.Critical, duration=3)
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/communicating.rst]: Why use "Notice" for the title with a Qgis.Critical message level?

Comment (ianturton) [reply source/docs/pyqgis_developer_cookbook/communicating.rst]: To show that it is not what causes the message to be an "error" - I've had students confused about this.

Comment (havatv) [reply source/docs/pyqgis_developer_cookbook/communicating.rst]: Perhaps "Ooops!" for the title and "The plugin is not working as it should" for the message?

Comment (ianturton) [reply source/docs/pyqgis_developer_cookbook/communicating.rst]: much better.

**Pull request merged**